Compiler/src/abstractinterpretation.jl

```diff
@@ -3497,6 +3497,17 @@ function merge_override_effects!(interp::AbstractInterpreter, effects::Effects,
     # It is possible for arguments (GlobalRef/:static_parameter) to throw,
     # but these will be recomputed during SSA construction later.
     override = decode_statement_effects_override(sv)
+    if override.consistent
+        m = interp.linfo.def
+        if isa(m, Method)
+            valid_worlds = WorldRange(m.primary_world, m.deleted_world)
+            if sv.world.this in valid_worlds
+                update_valid_age!(sv, valid_worlds)
```

Comment:
This logic is unsound to insert here. There is no legal use of `deleted_world` within Inference, as the value is not possible to synchronize with any source or edge. If this update was needed, it must have already occurred in the caller.

Reply:
Ok, I think it's fine to just leave the range open, because a deleted method will have the world age truncated from the ordinary lookup anyway.